skyflow/vault/_client.py

```diff
@@ -172,4 +174,83 @@ def update(self, updateInput, options: UpdateOptions = UpdateOptions()):
                                SkyflowErrorMessages.PARTIAL_SUCCESS, result, interface=interface)
         else:
             log_info(InfoMessages.UPDATE_DATA_SUCCESS.value, interface)
-            return result
\ No newline at end of file
+            return result
+
+    def delete_by_id(self, records: dict,options: DeleteOptions = DeleteOptions()):
+        interface = InterfaceName.DELETE_BY_ID.value
+        log_info(InfoMessages.DELETE_BY_ID_TRIGGERED.value, interface=interface)
+
+        self._checkConfig(interface)
+
+        self.storedToken = tokenProviderWrapper(
+            self.storedToken, self.tokenProvider, interface)
+        headers = {
+            "Authorization": "Bearer " + self.storedToken,
+            "sky-metadata": json.dumps(getMetrics())
+        }
+        error_list = []
+        result_list = []
+        errors = {}
+        result = {}
+        error = {}
+        try:
+            if not isinstance(records, dict) or "records" not in records:
+                error = {"error": {"code": SkyflowErrorCodes.INVALID_INPUT.value,
+                                   "description": SkyflowErrorMessages.RECORDS_KEY_NOT_FOUND_DELETE.value}}
+                return error
+            records_list = records["records"]
+            if not isinstance(records_list, list):
+                error.update({"error": {"code": SkyflowErrorCodes.INVALID_INPUT.value,
+                                        "description": SkyflowErrorMessages.INVALID_RECORDS_IN_DELETE.value}})
+                return error
+            elif len(records_list) == 0:
+                error = {"error": {"code": SkyflowErrorCodes.INVALID_INPUT.value,
+                                   "description": SkyflowErrorMessages.EMPTY_RECORDS_IN_DELETE.value}}
+                return error
+        except KeyError:
+            raise SkyflowError(SkyflowErrorCodes.INVALID_INPUT,
+                               SkyflowErrorMessages.RECORDS_KEY_ERROR, interface=interface)
+        try:
+            record_list = records["records"][0]['id']
+            if not isinstance(record_list, str):
+                error.update({"error": {"code": SkyflowErrorCodes.INVALID_INDEX.value,
+                                        "description": SkyflowErrorMessages.INVALID_ID_TYPE_DELETE.value}})
+                return error
+            elif record_list == "":
+                error.update({"error": {"code": SkyflowErrorCodes.INVALID_INPUT.value,
+                                        "description": SkyflowErrorMessages.EMPTY_ID_IN_DELETE.value}})
+                return error
+        except KeyError:
+            error.update({"error": {"code": SkyflowErrorCodes.INVALID_INDEX.value,
+                                    "description": SkyflowErrorMessages.IDS_KEY_ERROR.value}})
+            return error
+        try:
+            record_table = records["records"][0]['table']
```

Comment:
You shouldn't hard code the `index` of records here. It should be a loop over the records array. We need to validate every record in records not just the record at 0th index!